src/lib/y2network/hostname.rb

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) [2019] SUSE LLC
```

Comment:
Shall we defined it here or maybe better in yast2?

I already tried to open a discussion with this PR: https://github.com/yast/yast-yast2/pull/921

What in cases where we have a minimal image without yast2-network but still being able to handle hostname configuration, could be that the case?, as you commented, it has nothing to be with dns, so.. maybe we should have it in yast2.. although stil under y2network namespace...

Reply:
problem is that only UI we currently have is part of "yast2 lan". Moving it elsewhere was rejected. I'm currently not aware of any other use of hostname API